farm.py

```diff
@@ -0,0 +1,383 @@
+        self : class
+            required to change-by-reference members of the class
+            
+        ls : list
+            a list (of int) where each element represents a 'tree,' with
+            the element's value representing that specific tree's age
+            
+        see also:
+            makeTreesFromLand
+            
+        """
+        numTrees = len(ls)
+        
+        for i,e in enumerate(ls):
+            age = self.convertToRoundInt(e)
+            self.trees.append(age)
+            
+        self.totalTrees += numTrees
+        self.sowDensity = self.totalTrees * cuerdas
+        self.averageAgeOfTrees = stats.mean(self.trees)
+            
+    # this may be a subfunciton in the 'addYears' function
+    
+    def addCuerdas(self, cuerdas, ageOfTrees, treesPerCuerda = 2750):
+        # assign new variables to avoid confusion
+        # if user wants to add land without trees, set treesPerCuerda to zero (note: other variables in the class
+        # will reflect this change)
+        oldSowDens = self.sowDensity
+        oldCuerdas = self.totalCuerdas
+        newCuerdas = self.totalCuerdas + cuerdas
+        
+        self.sowDensity = (treesPerCuerda * (cuerdas / newCuerdas)) + (oldSowDens * (oldCuerdas / newCuerdas))
+        # calculate weighted avg and push new value to self.treesPerCuerda
+        
+        # acually update total cuerdas after to avoid confusion
+        self.totalCuerdas += cuerdas
+        
+                        
+        ageOfTrees = self.convertToRoundInt(ageOfTrees) 
+                        
+        # append the new trees to the
+        numTrees = self.convertToRoundInt(cuerdas * treesPerCuerda) # assure number of trees is iterable
+                        
+        for i in range(len(numTrees)):
+            self.trees.append(ageOfTrees)
+            
+        self.totalTrees += numTrees
+        self.averageAgeOfTrees = stats.mean(self.trees)
+        
+    
+    def setPruneTrees(self):
+        """
+        Takes in `pruneYear` bool, sets self.pruneYear to equiveleant value. 
+        
+        """
+        self.pruneYear = True
+        
+        # how much the total yield will increase after trees grow back from pruning (i.e. after pruneCount years)
+        self.pruneGrowth = 0.10 
+        
+        # current yield - (current yield * (this proportion * countyear)) = yield for that year.
+        self.pruneDelay = 0.20
+        # caution! The product of this proportion ^^^ and count year SHANT EXCEED 1.0 otherwise the 
+        # program will simulate impossibilities
+        
+        if pruneYear == True: # the if-else is set up in case the main function decides to pass this on ever iteration
+            self.pruneCount = 3
+            
+            # increase total yield-per-tree to 10% higher
+            self.harvestPerTree = self.harvestPerTree + (self.pruneGrowth * self.harvestPerTree)
+            
+        
+    def oneYear(self):
+        """
```

Comment:
we should talk about this. I think this is actually fine for now, but I see what happened. This implements the logic I was using (changing the values in the spreadsheet to track "state") rather than using this as an opportunity to do something better.

Reply:
really do need to think about the structure of the data we're trying to keep, how it is distinct from "current state" and how to go from data to "current state" effectively, reasonably.